### PR TITLE
Save And Restore Update

### PR DESCRIPTION
**Pull request by MichaelMCoates**
- Fix deregister bug in demo windows
- Properly restore child windows and tabbed windows
- Properly reassign the active tab in a tabset when restoring

### CODE REVIEW

File: src/demo/normalApp.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import {Identity} from 'hadouken-js-adapter';
 import {_Window} from 'hadouken-js-adapter/out/types/src/api/window/window';
 
-import {Layout, LayoutApp} from '../client/types';
-import {positionWindow} from '../provider/workspaces/utils';
+import {Layout, LayoutApp, WindowState} from '../client/types';
 
 export interface Workspace {
```
Comment (ryanpetersonOF) [inline src/demo/normalApp.ts]: Minor detail, can we move this below the imports?

File: src/demo/normalApp.ts [hunk 2]
```diff
@@ -18,22 +18,39 @@ const launchDir = location.href.slice(0, location.href.lastIndexOf('/'));
...
+            defaultWidth: 250 + 50 * i,
+            defaultLeft: 320 * (i % 3),
+            defaultTop: i > 2 ? 400 : 50,
+            saveWindowState: false,
+            frame,
+            name
+        });
+    }
+
     return win;
```
Comment (ryanpetersonOF) [inline src/demo/normalApp.ts]: You could do `return await fin.Window.create(...)` instead of assigning it to win

**Comment by tomrobertsOF** [comment]
Haven't been able to break it so far. 
It could be a bit confusing from a ux perspective that the basic snap windows don't save/restore, but I'm guessing that's because the layouts-manager deregisters itself.

File: src/demo/normalApp.ts [hunk 3]
```diff
@@ -1,27 +1,80 @@
...
+            await positionWindow(win);
+        }
+    });
+    await Promise.all(openAndPosition);
+    return layoutApp;
+}
+
+// Positions a window when it is restored.
+// Also given to the client to use.
 const positionWindow = async (win: WindowState) => {
```
[the remark this answers is not in the document]
Comment (MichaelMCoates) [reply src/demo/normalApp.ts]: Removed from LayoutsUI.ts

**Comment by tomrobertsOF** [comment]
Nice tidying pass. Looks much cleaner now.

**Pull request merged**